core/src/components/table/table-component-pagination.stories.tsx

```diff
@@ -175,61 +163,90 @@ const PaginationTemplate = ({
+  myData = [
+    {
+      "truck": "L-series",
+      "driver": "Sonya Bruce",
+      "country": "Brazil",
+      "mileage": 123987
+    },
+    {
+      "truck": "P-series",
+      "driver": "Guerra Bowman",
+      "country": "Sweden",
+      "mileage": 2000852
+    },
+    {
+      "truck": "G-series",
+      "driver": "Ferrell Wallace",
+      "country": "Germany",
+      "mileage": 564
+    },
+    {
+      "truck": "R-series",
+      "driver": "Cox Burris",
+      "country": "Spain",
+      "mileage": 1789357
+    },
+    {
+      "truck": "S-series",
+      "driver": "Montgomery Cervantes",
+      "country": "Croatia",
+      "mileage": 65
+    },
+    {
+      "truck": "L-series",
+      "driver": "Sheryl Nielsen",
+      "country": "Greece",
+      "mileage": 365784
+    },
+    {
+      "truck": "G-series",
+      "driver": "Benton Gomez",
+      "country": "France",
+      "mileage": 80957
+    },
+    {
+      "truck": "G-series",
+      "driver": "Benton Gomez",
+      "country": "France",
+      "mileage": 80957
+    }
+  ]
+  tableBody = document.querySelector('tds-table-body');
+  tableFooter = document.querySelector('tds-table-footer');
+  tableBody.bodyData = myData.slice(0,2);
+  rowsPerPage = 2;
 
-    /* ONLY WORKS IN THE CANVAS TAB. */
-    tableBody = document.querySelector('tds-table-body');
-    tableBody.bodyData = [
-      {
-        "truck": "L-series",
-        "driver": "Sonya Bruce",
-        "country": "Brazil",
-        "mileage": 123987
-      },
-      {
-        "truck": "P-series",
-        "driver": "Guerra Bowman",
-        "country": "Sweden",
-        "mileage": 2000852
-      },
-      {
-        "truck": "G-series",
-        "driver": "Ferrell Wallace",
-        "country": "Germany",
-        "mileage": 564
-      },
-      {
-        "truck": "R-series",
-        "driver": "Cox Burris",
-        "country": "Spain",
-        "mileage": 1789357
-      },
-      {
-        "truck": "S-series",
-        "driver": "Montgomery Cervantes",
-        "country": "Croatia",
-        "mileage": 65
-      },
-      {
-        "truck": "L-series",
-        "driver": "Sheryl Nielsen",
-        "country": "Greece",
-        "mileage": 365784
-      },
-      {
-        "truck": "G-series",
-        "driver": "Benton Gomez",
-        "country": "France",
-        "mileage": 80957
+
+    tableFooter.addEventListener('tdsPageChange', (event) => {
+      switch(event.detail.paginationValue){
+        case 1:
+          tableBody.bodyData = myData.slice(0,2);
+          console.log(1)
+          return;
+        case 2:
+          tableBody.bodyData = myData.slice(2,4);
+          console.log(2)
+          return;
+        case 3:
+          tableBody.bodyData = myData.slice(4,6);
+          console.log(3)
+          console.log( myData.slice(4,2))
+          return;
+        case 4:
+          tableBody.bodyData = myData.slice(6,8);
+          console.log(4)
+          return;
       }
-    ]
+    })
+
```

Comment:
Maybe remove the console.logs.

An alternative to the switch could be something like

```
const pageSize = 2;
const p = event.detail.paginationValue;
const start = (p - 1) * pageSize;
const end = p * pageSize;
tableBody.bodyData = myData.slice(start, end);
```

Reply:
Absolutely! Feel free to commit the suggested changes 🎈

Reply:
Not able to do it from github for whatever reason, but ill check it out

Reply:
done